Generalize H264 Docker image to work with Hobbyist or Voyager

The Janus-uStreamer proof of concept Docker images is currently hardcoded to use Voyager's uStreamer settings:

https://github.com/tiny-pilot/ansible-role-tinypilot/blob/48463dc5ef35bbdd373bb6c4a55600ceacd1a765/janus_ustreamer_docker/start.sh#L1

It should be pretty easy to make it work with Voyager or Hobbyist. I'd recommend adjusting start.sh so that uStreamer's parameters are specified by `"$@"` instead of hardcoded.

That way, when the user launches the Docker container, they can pass in parameters like:

```bash
docker run \
  --privileged \
  --network host \
  --name janus-ustreamer \
  mtlynch/ustreamer-janus:2022-02-02 \
  -f 30 \
  --host 127.0.0.1 \
  -p8001 \
  --h264-sink tinypilot::ustreamer::h264 \
  --h264-sink-rm \
  --h264-sink-mode 777 \
  --format uyvy \
  --encoder omx \
  --persistent \
  --dv-timings \
  --workers 3 \
  --drop-same-frames 30
```

That way, Docker passes the flags to start.sh, and start.sh passes those flags to uStreamer.

Then, we update the instructions in the README to show how to launch for Hobbyist or Voyager:

https://github.com/tiny-pilot/tinypilot/tree/experimental/h264#run

# Update

I'm running into this error when building the docker image:
```bash
$ docker build -t janus-ustreamer .
...
#9 154.0 CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
#9 154.0   Could NOT find ZLIB (missing: ZLIB_LIBRARY) (found version "1.2.11")
#9 154.0 Call Stack (most recent call first):
#9 154.0   /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
#9 154.0   /usr/share/cmake-3.16/Modules/FindZLIB.cmake:115 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
#9 154.0   CMakeLists.txt:914 (find_package)
#9 154.0
#9 154.0
#9 154.1 -- Configuring incomplete, errors occurred!
#9 154.1 See also "/libwebsockets/build/CMakeFiles/CMakeOutput.log".
#9 154.1 See also "/libwebsockets/build/CMakeFiles/CMakeError.log".
------
executor failed running [/bin/sh -c git clone https://libwebsockets.org/repo/libwebsockets &&     cd libwebsockets &&     mkdir build &&     cd build &&     cmake -DLWS_MAX_SMP=1 -DLWS_WITHOUT_EXTENSIONS=0 -DCMAKE_INSTALL_PREFIX:PATH=/usr -DCMAKE_C_FLAGS="-fpic" .. &&     make && sudo make install]: exit code: 1
```

It seems to be related to this bug: https://gitlab.kitware.com/cmake/cmake/-/issues/20127

The workaround seems to be: https://discourse.cmake.org/t/cant-find-zlib-missing-zlib-library-ubuntu-20-cmake/2610/2
```bash
cmake ... -DZLIB_LIBRARY=/usr/zlib/lib/libz.so ...
```


However, after I apply the workaround, the build seems to be failing silently?

```bash
...
#9 255.2 You are about to be asked to enter information that will be incorporated
#9 255.2 into your certificate request.
#9 255.2 What you are about to enter is what is called a Distinguished Name or a DN.
#9 255.2 There are quite a few fields but you can leave some blank
#9 255.2 For some fields there will be a default value,
#9 255.2 If you enter '.', the field will be left blank.
#9 255.2 -----
#9 255.2 Country Name (2 letter code) [AU]:State or Province Name (full name) [Some-State]:Locality Name (eg, city) []:Organization Name (eg, company) [Internet Widgits Pty Ltd]:Organizational Unit Name (eg, section) []:Common Name (e.g. server FQDN or YOUR name) []:Email Address []:SUCCESSFULLY generated SSL certificate
#9 255.7 DIR /libwebsockets/build CMP /libwebsockets/cmake
#9 256.5 -- Configuring incomplete, errors occurred!
#9 256.5 See also "/libwebsockets/build/CMakeFiles/CMakeOutput.log".
#9 256.5 See also "/libwebsockets/build/CMakeFiles/CMakeError.log".
------
executor failed running [/bin/sh -c git clone https://libwebsockets.org/repo/libwebsockets &&     cd libwebsockets &&     mkdir build &&     cd build &&     cmake -DLWS_MAX_SMP=1 -DLWS_WITHOUT_EXTENSIONS=0 -DCMAKE_INSTALL_PREFIX:PATH=/usr -DCMAKE_C_FLAGS="-fpic" -DZLIB_LIBRARY=/usr/local/zlib/lib/libz.so -DZLIB_INCLUDE_DIR=/usr/local/zlib/include .. &&     make && sudo make install]: exit code: 1
```

I'll need figure out how to get access to the cmake logs inside docker, or I can just try run these build commands directly on a device instead. I'll keep investigating.

--- a/janus_ustreamer_docker/start.sh
+++ b/janus_ustreamer_docker/start.sh
@@ -1,2 +1,2 @@
-/ustreamer/ustreamer -f 30 --host 127.0.0.1 -p8001 --h264-sink tinypilot::ustreamer::h264 --h264-sink-rm --h264-sink-mode 777 --format uyvy --encoder omx --persistent --dv-timings --workers 3 --drop-same-frames 30 &
+/ustreamer/ustreamer "$@" &
 /opt/janus/bin/janus -F /opt/janus/lib/janus/configs/ -P /opt/janus/lib/janus/plugins/

--- a/tasks/main.yml
+++ b/tasks/main.yml
@@ -7,12 +7,16 @@
 #  tags:
 #    - ustreamer
 
+- name: Populate service facts
+  ansible.builtin.service_facts:
+
 - name: disable uStreamer
   systemd:
     name: ustreamer
     daemon_reload: yes
     enabled: no
     state: stopped
+  when: 'ustreamer' in services
 
 - name: install nginx
   import_tasks: nginx.yml

--- a/tasks/main.yml
+++ b/tasks/main.yml
@@ -16,7 +16,7 @@
     daemon_reload: yes
     enabled: no
     state: stopped
-  when: 'ustreamer' in services
+  when: "'ustreamer' in services"
 
 - name: install nginx
   import_tasks: nginx.yml

